src/hooks/useTimer.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+
+// TODO: Abstract this as a props
+const mintDate = new Date('2022-05-23T00:00:00');
+
+export default function useTimer() {
+  const currentTime = Date.now();
+
+  const [countDown, setCountDown] = useState(mintDate.getTime() - currentTime);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setCountDown(mintDate.getTime() - Date.now());
+    }, 1000);
+
+    return () => clearInterval(interval);
+  }, [countDown]);
+
+  const days = Math.floor(countDown / (1000 * 60 * 60 * 24))
+    .toString()
+    .padStart(2, '0');
+  const hours = Math.floor((countDown % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60))
+    .toString()
+    .padStart(2, '0');
+  const minutes = Math.floor((countDown % (1000 * 60 * 60)) / (1000 * 60))
+    .toString()
+    .padStart(2, '0');
+  const seconds = Math.floor((countDown % (1000 * 60)) / 1000)
+    .toString()
+    .padStart(2, '0');
+
+  const hasEnded = countDown <= 0;
+
+  return {
+    days,
+    hours,
+    minutes,
+    seconds,
+    hasEnded,
+  };
+}
```

Comment:
nit: doesn't block this PR, but maybe we can write some unit tests for this? wanna make sure things still look good if there's just 1 minute left (e.g. `00:00:01:00`)

Reply:
also nit: judging by how the values are used, we could return:

```js
  return {
    timestamp: `${days}:${hours}:${minutes}:${seconds}`,
    hasEnded,
  };
```

instead of 

```js
  return {
    days,
    hours,
    minutes,
    seconds,
    hasEnded,
  };
```

Reply:
Definitely! I can write it in the next PR